roles/alloy/tasks/install.yml

```
@@ -26,22 +26,13 @@
   become: true
   register: download_result
 
-- name: Remove existing alloy installation directory
+- name: Remove existing alloy binary
   ansible.builtin.file:
-    path: "{{ installation_dir }}"
+    path: "{{ installation_dir }}/alloy-{{ version }}"
```

AFAIK the binary name does not contain a version string.
Perhaps you meant to put the architecture in the name?

```suggestion
    path: "{{ installation_dir }}/alloy-linux-{{ arch }}"
```

---

Good catch!